File: Algorithms/Implementation/Between Two Sets.php
<?php

/*
 * Complete the 'getTotalX' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts following parameters:
 *  1. INTEGER_ARRAY a
 *  2. INTEGER_ARRAY b
 */

function getTotalX($a, $b) {
    # Write your code here
    $count = 0;
    for($i = min($a); $i <= max($b); $i++) {
        $flag = 1;
        foreach($a as $val) {
            if($i % $val != 0) {
                $flag = 0;
                break;
            }
        }
        if($flag == 1) {
            foreach($b as $val) {
                if($val % $i != 0) {
                    $flag = 0;
                    break;
                }
            }
        }
        if($flag == 1) {
            $count++;
        }
    }
    return $count;
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");
$first_multiple_input = explode(' ', rtrim(fgets(STDIN)));

$n = intval($first_multiple_input[0]);
$m = intval($first_multiple_input[1]);

$arr_temp = rtrim(fgets(STDIN));
$arr = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));

$brr_temp = rtrim(fgets(STDIN));
$brr = array_map('intval', preg_split('/ /', $brr_temp, -1, PREG_SPLIT_NO_EMPTY));

$total = getTotalX($arr, $brr);

fwrite($fptr, $total . "\n");
fclose($fptr);

File: Algorithms/Implementation/Between Two Sets.py
import os


#
# Complete the 'getTotalX' function below.
#
# The function is expected to return an INTEGER.
# The function accepts following parameters:
#  1. INTEGER_ARRAY a
#  2. INTEGER_ARRAY b
#

def getTotalX(a, b):
    # Write your code here
    total = 0
    for i in range(max(a), min(b) + 1):
        if all(i % j == 0 for j in a) and all(j % i == 0 for j in b):
            total += 1
    return total


if __name__ == '__main__':

    fptr = open(os.environ['OUTPUT_PATH'], 'w')
    first_multiple_input = input().rstrip().split()

    n = int(first_multiple_input[0])
    m = int(first_multiple_input[1])

    arr = list(map(int, input().rstrip().split()))
    brr = list(map(int, input().rstrip().split()))

    total = getTotalX(arr, brr)

    fptr.write(str(total) + '\n')
    fptr.close()
